module #2. REST API Basics/rest-api-basic/core/src/main/java/com/epam/esm/model/GiftCertificate.java

```diff
@@ -0,0 +1,32 @@
+package com.epam.esm.model;
+
+
+import lombok.*;
+
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.Set;
+
+@AllArgsConstructor
+@NoArgsConstructor
+@Data
+@Builder
+@EqualsAndHashCode
+
+public class GiftCertificate {
+
+    private Long id;
+
+    private String name;
+
+    private String description;
+
+    private Double price;
+
+    private Integer duration;
+
+    private LocalDateTime createDate;
+    private LocalDateTime lastUpdateDate;
+    @Builder.Default
+    private Set<Tag> tags = new HashSet<>();
```

Comment:
It is better to exclude it from Equals and Hash code.

Try this please, @EqualsAndHashCode.Exclude